src/playback-watcher.js

```diff
@@ -122,11 +122,23 @@ export default class PlaybackWatcher {
   waiting_() {
     let seekable = this.seekable();
     let currentTime = this.tech_.currentTime();
+    let buffered = this.tech_.buffered();
 
     if (this.tech_.seeking() || this.timer_ !== null) {
       return;
     }
 
+    if (this.pastLiveWindow_(seekable, currentTime. buffered)) {
+      let seekableEnd = seekable.end(seekable.length - 1);
+
+      // Through slow playlist requests or some other means, we managed to find ourselves
+      // off the front of the live window. It doesn't matter how we got here, but we need
+      // to get back to our safe seekable range.
+      this.logger_(`Fell past live window at time ${currentTime}. Seeking to ` +
+                   `seekable end ${seekableEnd}`);
+      this.tech_.setCurrentTime(seekableEnd);
```

Comment:
Looks like we're doing the same thing for falling out of the live window, maybe the logic should be combined.